src/navier.f90

```diff
@@ -1280,5 +1289,266 @@ subroutine tbl_flrt (ux1,uy1,uz1)
     enddo
 
   end subroutine tbl_flrt
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  !!
+  !!  subroutine: pipe_bulk / pipe_bulk_u / pipe_bulk_phi
+  !!      AUTHOR: Rodrigo Vicente Cruz
+  !! DESCRIPTION: Correction of pipe's bulk velocity (constant 
+  !!              flow rate) and bulk temperature
+  !!
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  !********************************************************************
+  !
+  subroutine pipe_bulk(ux,uy,uz,ep)
+  !
+  !********************************************************************
+
+    use param
+    use decomp_2d
+    use variables
+    use var,  only: phi1
+
+    implicit none
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3))  :: ux,uy,uz,ep
+    integer                                             :: is
+
+    !Bulk temperature correction
+    if (numscalar.ne.0) then
+        do is=1,numscalar
+            call pipe_bulk_phi(phi1(:,:,:,is),ux,ep,is,one)
+        enddo
+    endif
+
+    !Bulk velocity correction
+    call pipe_bulk_u(ux,uy,uz,ep,one)
+
+  end subroutine pipe_bulk
+  !********************************************************************
+  !
+  subroutine pipe_bulk_u(ux,uy,uz,ep,ub_constant)
+  !
+  !********************************************************************
+
+    use decomp_2d
+    use variables
+    use param
+    use var
+    use ibm_param, only: ra
+    use MPI
+
+    implicit none
+    !INPUTS
+    real(mytype),intent(inout),dimension(xsize(1),xsize(2),xsize(3))    :: ux,uy,uz
+    real(mytype),intent(in   ),dimension(xsize(1),xsize(2),xsize(3))    :: ep
+    real(mytype),intent(in   )                                          :: ub_constant !bulk velocity value
+    !LOCALS
+    real(mytype)                                                        :: qm,qmm      !flow rate
+    real(mytype)                                                        :: ym,zm,yc,zc,r
+    real(mytype)                                                        :: ncount,ncountt
+    integer                                                             :: j,i,k,code
+    character(len=30)                                                   :: filename
+    integer, save                                                       :: local_io_unit=-1
+
+    yc = yly/two
+    zc = zlz/two
+
+    if (itime.eq.ifirst.and.nrank.eq.0) then
```

Comment:
I suspect testing `itime` is not sufficient in case of sub-iterations (RK time scheme). A workaround could be to replace
```
if (itime.eq.ifirst.and.nrank.eq.0) then
```
with
```
if (local_io_unit.eq.-1and.nrank.eq.0) then
```

Reply:
Thank you for seeing this issue, it has been fixed in ffc0c3c6b1f583d70523b2fea8ce541392590d47.